### PR TITLE
Added date encode and decode instances

### PR DESCRIPTION
**Pull request by cryogenian**

### CODE REVIEW

File: src/Data/Argonaut/Instances/Datetime.purs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  } else {
+    return nothing;
+  }
+}
+""" :: forall a. Fn3 Json (a -> Maybe a) (Maybe a) (Maybe JSDate)
+
+json2JSDate :: Json -> Maybe JSDate
+json2JSDate json = runFn3 json2JSDateImpl json (Just) (Nothing)
+
+instance decodeJsonJSDate :: DecodeJson JSDate where
```
Comment (jdegoes) [inline src/Data/Argonaut/Instances/Datetime.purs]: These are currently orphan instances. They should be defined in the `EncodeJson` and `DecodeJson` modules to make sure they are not orphans.

**Comment by cryogenian** [comment]
ok, I've moved them to EncodeJson and DecodeJson modules

File: src/Data/Argonaut/Decode.purs [hunk 2]
```diff
@@ -107,3 +109,30 @@ module Data.Argonaut.Decode
 
   -- arrayMembersL :: forall a. (DecodeJson a, EncodeJson a) => IndexedTraversalP JNumber Json a
   -- arrayMembersL = decodeL >>> traversed >>> arrayL
+
+
+  foreign import json2JSDateImpl """
+  function json2JSDateImpl(json, just, nothing) {
+    if (json && json.constructor && json.constructor == Date) {
```
Comment (jdegoes) [inline src/Data/Argonaut/Decode.purs]: Sorry, I should have looked at this native code more closely before -- How is this a valid JSON encoding (i.e. can survive `parse` / `stringify`)?

**Comment by cryogenian** [comment]
Hmph... I think that this will not be usefull

**Pull request closed without merging**